### ISSUE TITLE
Issue With Logging On AppModule Initialization

### ISSUE THREAD

**Issue by gpresland** [nestjs/nest#1081]
[ ] Regression
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request
</code></pre>

This is somewhat related to issue #926.

## Current behavior

Disabling logging or using a custom logger is bypassed/ignored by errors that occur during dependency injection i.e. `AppModule` initialization.

## Expected behavior

Logger should be set up before any dependency injection or bootstrapping. That way any/all error are sent through the custom logger or are ignored if logging is false.

## Minimal reproduction of the problem with instructions

Use a custom logger in your _main.ts_ `bootstrap`, or even disable logging.

**main.ts**
```typescript
...
async function bootstrap() {
  const app = await NestFactory.create(AppModule, {
    logger: false
  });
  ...
```

Perform an import in _app.module.ts_ which will throw an error, e.g. import `TypeOrmModule` with bad connection parameters.

Example: TypeORM fails to initialize because of bad connection parameters, it will throw an error which Nest will catch and put out into the console with its own logger and include the stack trace *even though logging is disabled or is told to use a custom logger!*.

**app.module.ts**
```typescript
@Module({
  imports: [
    TypeOrmModule.forRoot({
   // Bad parameters that will fail to connect, thus throwing an error.
    ...
```

## What is the motivation / use case for changing the behavior?

- Ability to catch **all** application errors so that the developer can choose how to output them to the console, or allows him to do his own custom logging method as to where/how to save errors.
- Ability to hide **all** application errors.
- Ability to hide stack traces.
- If all logging output is done with one logger in one format, logs can be more easily parsed.

## Environment

Nest version: 5.4.2

**Comment by kamilmysliwiec** [comment]
I have explained it here: https://github.com/nestjs/nest/issues/926

**Comment by gpresland** [comment]
@kamilmysliwiec Yeah, I referenced that issue above. I am on version 5.4.2 which you implemented a fix for that issue in 5.2.0. The current issue is, is that the Nest logger will still output errors in its custom logger if a module throws an error. Regardless of whether you're using a custom logger or have logging turned off. I have tried the solution in #926.
